### PR TITLE
Group trainer call methods as functions

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Remove code from `trainer.py` that doesn't need to be there.

cc @justusschock @awaelchli @borda

### CODE REVIEW

File: src/lightning/pytorch/core/module.py [hunk 1]
```diff
@@ -42,6 +42,7 @@
 from lightning.pytorch.core.optimizer import LightningOptimizer
 from lightning.pytorch.core.saving import ModelIO
 from lightning.pytorch.loggers import Logger
+from lightning.pytorch.trainer import call
```
Comment (Borda) [inline src/lightning/pytorch/core/module.py:45]:
```suggestion
from lightning.pytorch.trainer import _calls
```
 seems all func there protected so let make whole module protected

File: src/lightning/pytorch/loops/dataloader/evaluation_loop.py [hunk 2]
```diff
@@ -26,6 +26,7 @@
 from lightning.pytorch.loops.epoch import _EvaluationEpochLoop
 from lightning.pytorch.loops.fetchers import _DataFetcher
 from lightning.pytorch.loops.utilities import _select_data_fetcher, _set_sampler_epoch
+from lightning.pytorch.trainer import call
```
Comment (Borda) [inline src/lightning/pytorch/loops/dataloader/evaluation_loop.py:29]: Then also rather import each function then this module call which may confuse as being another object...

**Review by Borda — APPROVED** [review]

File: src/lightning/pytorch/loops/dataloader/evaluation_loop.py (hunk 2 above)
Comment (carmocca) [reply src/lightning/pytorch/loops/dataloader/evaluation_loop.py:29]: Doing it like this follows the pattern started by William at https://github.com/Lightning-AI/lightning/pull/14871 and https://github.com/Lightning-AI/lightning/pull/14862

**Review by awaelchli — APPROVED** [review]
PR LGTM, but not happy about the pattern that was started prior to this :)

File: src/lightning/pytorch/core/module.py (hunk 1 above)
Comment (carmocca) [reply src/lightning/pytorch/core/module.py:45]: I'll leave this for a follow-up if we want to do it. We don't have any other protected modules and I don't want to stray away from the current pattern used in the trainer unless there's consensus

**Comment by Borda** [comment]
> PR LGTM, but not happy about the pattern that was started prior to this :)

the same here :(

**Pull request merged**